packages/amplication-server/src/core/project/project.resolver.ts

```diff
@@ -92,9 +103,75 @@ export class ProjectResolver {
   )
   @InjectContextValue(InjectableOriginParameter.UserId, "data.user.connect.id")
   async commit(@Args() args: CreateCommitArgs): Promise<Commit | null> {
+    const projectId = args.data.project.connect.id;
+
+    await this.validateProject(projectId);
+
     return this.projectService.commit(args);
   }
 
+  async validateProject(projectId: string): Promise<void> {
+    const project = await this.projectService.findUnique({
+      where: { id: projectId },
+    });
+    const workspace = await this.workspaceService.getWorkspace({
+      where: { id: project.workspaceId },
+    });
```

Comment:
Why not do that as one request, and update the DTO to include the workspace?

Reply:
Take a look at `findUnique` func, input supports only `where` clause, there is no `include` field. I agree that it would be nice to make most of these operations by one request with joins, but that's the api that we have right now.

Reply:
We can add to the API the include field in the args dto

Reply:
We can, but the feature is not about refactoring data services, the scope is clear - adding paywall limitation. Again, I agree that would be better, but it has to be done separately.